web_src/js/components/DiffCommitSelector.vue

```diff
@@ -0,0 +1,196 @@
+        this.isLoading = false;
+      }
+    },
+    /** Load the commits to show in this dropdown */
+    async fetchCommits() {
+      const resp = await fetch(`${this.issueLink}/commits/list`);
+      const results = await resp.json();
+      this.commits.push(...results.commits);
+      this.commits.reverse();
+      this.lastReviewCommitSha = results.last_review_commit_sha || null;
+      Object.assign(this.locale, results.locale);
+    },
+    showAllChanges() {
+      window.location = `${this.issueLink}/files${this.queryParams}`;
+    },
+    /** Called when user clicks on since last review */
+    changesSinceLastReviewClick() {
+      window.location = `${this.issueLink}/files/${this.lastReviewCommitSha}..${this.commits.at(-1).id}${this.queryParams}`;
+    },
+    /** Clicking on a single commit opens this specific commit */
+    commitClicked(commitId) {
+      window.location = `${this.issueLink}/commits/${commitId}${this.queryParams}`;
+    },
+    /**
+     * When a commit is clicked with shift this enables the range
+     * selection. Second click (with shift) defines the end of the
+     * range. This opens the diff of this range
+     * Exception: first commit is the first commit of this PR. Then
+     * the diff from beginning of PR up to the second clicked commit is
+     * opened
+     */
+    commitClickedShift(commit) {
+      this.hoverActivated = !this.hoverActivated;
+      commit.selected = true;
+      // Second click -> determine our range and open links accordingly
+      if (!this.hoverActivated) {
+        // find all selected commits and generate a link
+        if (this.commits[0].selected) {
+          // first commit is selected - generate a short url with only target sha
+          const lastCommitIdx = this.commits.findLastIndex((x) => x.selected);
+          if (lastCommitIdx === this.commits.length - 1) {
+            // user selected all commits - just show the normal diff page
+            window.location = `${this.issueLink}/files${this.queryParams}`;
+          } else {
+            window.location = `${this.issueLink}/files/${this.commits[lastCommitIdx].id}${this.queryParams}`;
+          }
+        } else {
+          const start = this.commits[this.commits.findIndex((x) => x.selected) - 1].id;
+          const end = this.commits.findLast((x) => x.selected).id;
+          window.location = `${this.issueLink}/files/${start}..${end}${this.queryParams}`;
+        }
+      }
+    },
+  }
+};
+</script>
+<style scoped>
+  .selected {
+    background-color: var(--color-secondary-active) !important;
+  }
+
+  .info {
+    display: inline-block;
+    padding: 7px 14px !important;
+    line-height: 1.4;
+    width: 100%;
+  }
+
+  .ui.dropdown .menu .menu.is-loading {
+    left: 0;
+    height: 200px;
+    width: 200px !important;
+  }
+
+  .ui.dropdown .menu .menu.is-loading::after {
+    display: block !important; /* to override fomantic rule .ui.dropdown .menu .menu:after {display: none} */
+  }
```

Comment:
I think it could avoid conflicting with Fomantic dropdown styles, use a separate element for "is-loading".

ps: there are some other nits in my mind, if you don't mind, could I edit your PR directly?

https://github.com/go-gitea/gitea/blob/79995a8194235ee994a1ce3cbe5eaa11358db885/web_src/js/components/RepoBranchTagSelector.vue#L41

Reply:
Please feel free to edit the pr directly!

Reply:
> maybe you got an idea / solution for the inaccessibiltity with a keyboard?

I have thought about it. IMO it needs a separate framework level support in Vue. At the moment, Gitea's frontend lacks an a11y friendly Vue UI framework.